Automattic/jetpack
# My Jetpack: Add missing TOS in the product interstitials.

This PR fixes missing Term of service texts in some of the product interstitial screens. Note that there are guidelines on where TOS should be displayed. Refer to this link for the discussion: p1HpG7-jRz

<img width="1139" alt="Screen Shot 2023-03-31 at 4 17 10 PM" src="https://user-images.githubusercontent.com/56598660/229065166-69c84f03-0f92-44d9-a8e0-9dbfb2f202ea.png">
<img width="1082" alt="Screen Shot 2023-03-31 at 4 18 07 PM" src="https://user-images.githubusercontent.com/56598660/229065453-f3fa40ee-3c4e-4eeb-9cc9-fba98e81f91e.png">
<img width="1114" alt="Screen Shot 2023-03-31 at 4 18 24 PM" src="https://user-images.githubusercontent.com/56598660/229065847-ec5a5b92-1399-47e3-8ff7-989bd8aebc2d.png">


Fixes #29610

## Proposed changes:
* Update **Product interstitial** component in **My Jetpack** package to render `TermOfService` component.
* Update **Product description** component in **Jetpack** plugin to render `TermOfService` component.

### Other information:


## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
* Spin up a  local WP site and checkout in this branch (`add/missing-interstitial-tos`).
    * Build both my jetpack package and jetpack plugin by running the command below
```
jetpack build packages/my-jetpack
jetpack build plugins/jetpack
```
* Or you can Spin up a JN site with Jetpack beta pointing to this branch (`add/missing-interstitial-tos`)
* Connect your site w/ your WPCOM account (You will need jurassic tube to do this if you are running a local WP site)
* Visit all screens below and confirm that the TOS is rendered in the correct place.

URL |  requires connection?
---   | ---
/wp-admin/admin.php?page=my-jetpack#/add-anti-spam   | No
/wp-admin/admin.php?page=my-jetpack#/add-backup   | No
/wp-admin/admin.php?page=my-jetpack#/add-boost  | No
/wp-admin/admin.php?page=my-jetpack#/add-crm  | No
/wp-admin/admin.php?page=my-jetpack#/add-extras  | No
/wp-admin/admin.php?page=my-jetpack#/add-protect  | No
/wp-admin/admin.php?page=my-jetpack#/add-scan   | No
/wp-admin/admin.php?page=my-jetpack#/add-social  | No
/wp-admin/admin.php?page=my-jetpack#/add-search   | No
/wp-admin/admin.php?page=my-jetpack#/add-videopress  | No
/wp-admin/admin.php?page=jetpack#/product/akismet   | ✅  
/wp-admin/admin.php?page=jetpack#/product/backup   | ✅ 
/wp-admin/admin.php?page=jetpack#/product/scan   | ✅ 
/wp-admin/admin.php?page=jetpack#/product/search  | ✅ 
/wp-admin/admin.php?page=jetpack#/product/security  | ✅ 
/wp-admin/admin.php?page=jetpack#/product/videopress  | ✅
